### ISSUE TITLE
Object_name for blob table

### ISSUE THREAD

**Issue by lavvy** [crate/crate#6471]
Use cases
Making blob store a an object store

Feature
An additional non strict column in the blob table wont hurt unless there is a reason for not.
Currently there are sha column and last updated.
i think it would be nice to make 3 columns =  object_name, last_updated and sha.

Advantages
It will ease in better human identification and management of blobs. It will create a better interface to build an object storage on crate.

**Comment by joemoe** [comment]
hi @lavvy, thanks for your suggestion.
i'll add the enhancement label, but i guess we won't add that soon.

**Comment by junphine** [comment]
this feature is very useful. server side compute sha will simplify client usage for upload file with file name.

**Comment by MirtoBusico** [comment]
In my use case (see [What are the best practices for managing blob’s metadata?](https://community.cratedb.com/t/what-are-the-best-practices-for-managing-blobs-metadata/1947) ) I need to upload and download some files that are considered attachments to documents.

I need to do a full text search on a file logical name and/or a full text search on an abstract of the file (giving that blobs are not indexable).
Then the selected blobs have to be downloaded restoring the original file name and without losing logical name and abstract.

**TL;DR**

As a demo, I have these two tables
```
cr> SELECT * FROM BLOB.attachments;
+------------------------------------------+---------------+
| digest                                   | last_modified |
+------------------------------------------+---------------+
| acfa51b963b5a1fb9eecaf7954ae39c9bea736cc | 1735995025468 |
| 135b6d193c9acba8ea180dea7424b863ce52858b | 1735994937133 |
| f83be20cb0439e5008242ec7a01cf1fe168f413b | 1735994988697 |
| 9a3993e2ecbbb3b1ea1435650cedd7fec2454e73 | 1735995055568 |
+------------------------------------------+---------------+
SELECT 4 rows in set (0.020 sec)
cr> SELECT * FROM attachments_meta;
+------------------------------------------+-----------------------------------------------+-----------------------------------------------------------------------------------------+----------------------------------------------+
| digest                                   | name                                          | description                                                                             | filename                                     |
+------------------------------------------+-----------------------------------------------+-----------------------------------------------------------------------------------------+----------------------------------------------+
| 135b6d193c9acba8ea180dea7424b863ce52858b | Insert CPU IN motherboard socket              | Short video ON how TO INSERT a CPU IN the motherboard socket                            | 11537353-hd_1920_1080_30fps.mp4              |
| f83be20cb0439e5008242ec7a01cf1fe168f413b | Impress Guide v. 24.2                         | LibreOffice Impress Guide version 24.2. A manual FOR creating presentations             | IG24-ImpressGuide.pdf                        |
| acfa51b963b5a1fb9eecaf7954ae39c9bea736cc | ROG Maximus Z790 Hero BTF - Quick START guide | The quick START guide FOR the Asus Republic Of Gamers Maximus Z790 Hero BTF motherboard | Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf |
| 9a3993e2ecbbb3b1ea1435650cedd7fec2454e73 | Top VIEW of ROG CROSSHAIR X670E HERO          | The top VIEW image of the Asus Republic Of Gamers CROSSHAIR X670E HERO motherboard      | ROG_CROSSHAIR_X670E_HERO_h732.png            |
+------------------------------------------+-----------------------------------------------+-----------------------------------------------------------------------------------------+----------------------------------------------+
SELECT 4 rows in set (0.020 sec)
cr>
```

Using this content, I was able to create a script that does a full text search on the "description" column to find attachments that contain the "asus" word, and download the found blobs and chreate a json file containing the metadata.

The script
```
#!/bin/bash

echo "usage: ../getattachment "

SCRIPTPATH=/usr/share/nginx/html/data/attachments/

MATCHFILTER='asus'
#MATCHFILTER='btf'

echo =
OUTPUT=`
curl -sS --user system:S9xsloAiGVvZ7iqaDQjtAzasmiFedEYlc1ajapZWDikhPbZTkj -H 'Content-Type: application/json' -X POST '10.43.47.153:4200/_sql' -d@- <<- EOF
{
    "stmt":
    "SELECT * FROM attachments_meta
    WHERE MATCH(description, '$MATCHFILTER'); "
}
EOF
`

NUMROW=`echo $OUTPUT | jq -r '.rowcount' `
NUMR=$((NUMROW - 1))


COLS=`echo $OUTPUT | jq -r '.cols' `
COL1=`echo $COLS | jq -r '.[0]' `
COL2=`echo $COLS | jq -r '.[1]' `
COL3=`echo $COLS | jq -r '.[2]' `
COL4=`echo $COLS | jq -r '.[3]' `

ROWS=`echo $OUTPUT | jq -r '.rows' `

for x in $(eval echo "{0..$NUMR}");
do
    NUM="".[$x]
    echo
    echo ==========================================================================================
    ROW=`echo $ROWS | jq -r $NUM `
    echo ROW $x = $ROW
    echo

    VAL1=`echo $ROW | jq -r '.[0]' `
    VAL2=`echo $ROW | jq -r '.[1]' `
    VAL3=`echo $ROW | jq -r '.[2]' `
    VAL4=`echo $ROW | jq -r '.[3]' `
    echo VAL4 = $VAL4
    METAFILE=$VAL4.json
    echo -n {\"$COL1\": \"$VAL1\", \"$COL2\": \"$VAL2\", \"$COL3\": \"$VAL3\", \"$COL4\": \"$VAL4\"} > $METAFILE

    URL=10.43.47.153:4200/_blobs/attachments/$VAL1
    curl -sS --output $VAL4 $URL --user system:S9xsloAiGVvZ7iqaDQjtAzasmiFedEYlc1ajapZWDikhPbZTkj

done
```

The output
```
root@crashtest-9dfdd4b8c-ggs8h:/usr/share/nginx/html/data/attachments/temp# ls -lh
total 0
root@crashtest-9dfdd4b8c-ggs8h:/usr/share/nginx/html/data/attachments/temp# ../getattachment 
usage: ./getattachment 
=

==========================================================================================
ROW 0 = [ "acfa51b963b5a1fb9eecaf7954ae39c9bea736cc", "ROG Maximus Z790 Hero BTF - Quick START guide", "The quick START guide FOR the Asus Republic Of Gamers Maximus Z790 Hero BTF motherboard", "Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf" ]

VAL4 = Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf

==========================================================================================
ROW 1 = [ "9a3993e2ecbbb3b1ea1435650cedd7fec2454e73", "Top VIEW of ROG CROSSHAIR X670E HERO", "The top VIEW image of the Asus Republic Of Gamers CROSSHAIR X670E HERO motherboard", "ROG_CROSSHAIR_X670E_HERO_h732.png" ]

VAL4 = ROG_CROSSHAIR_X670E_HERO_h732.png
root@crashtest-9dfdd4b8c-ggs8h:/usr/share/nginx/html/data/attachments/temp# ls -lh
total 1.9M
-rw-r--r-- 1 root root 1.8M Jan  6 20:38 Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf
-rw-r--r-- 1 root root  277 Jan  6 20:38 Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf.json
-rw-r--r-- 1 root root    0 Jan  6 20:38 ROG_CROSSHAIR_X670E_HERO_h732.png
-rw-r--r-- 1 root root  252 Jan  6 20:38 ROG_CROSSHAIR_X670E_HERO_h732.png.json
root@crashtest-9dfdd4b8c-ggs8h:/usr/share/nginx/html/data/attachments/temp# 
```

The json files content
```
root@crashtest-9dfdd4b8c-msmqn:/usr/share/nginx/html/data/attachments/temp# cat Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf.json |jq
{
  "digest": "acfa51b963b5a1fb9eecaf7954ae39c9bea736cc",
  "name": "ROG Maximus Z790 Hero BTF - Quick START guide",
  "description": "The quick START guide FOR the Asus Republic Of Gamers Maximus Z790 Hero BTF motherboard",
  "filename": "Q23155_ROG_MAXIMUS_Z790_HERO_BTF_QSG_WEB.pdf"
}
root@crashtest-9dfdd4b8c-msmqn:/usr/share/nginx/html/data/attachments/temp# cat ROG_CROSSHAIR_X670E_HERO_h732.png.json |jq
{
  "digest": "9a3993e2ecbbb3b1ea1435650cedd7fec2454e73",
  "name": "Top VIEW of ROG CROSSHAIR X670E HERO",
  "description": "The top VIEW image of the Asus Republic Of Gamers CROSSHAIR X670E HERO motherboard",
  "filename": "ROG_CROSSHAIR_X670E_HERO_h732.png"
}
root@crashtest-9dfdd4b8c-msmqn:/usr/share/nginx/html/data/attachments/temp# 
```
